src/imitation/rewards/discrim_net.py

```diff
@@ -308,33 +354,67 @@ def act_ph(self):
-
-    hid_sizes = self.hid_sizes
-    if hid_sizes is None:
-      hid_sizes = (32, 32)
-    discrim_mlp = util.build_mlp(hid_sizes=hid_sizes)
-    discrim_logits = util.sequential(inputs, discrim_mlp)
-
-    return discrim_mlp, discrim_logits
-
-  def build_policy_train_reward(self) -> tf.Tensor:
-    train_reward = -tf.log_sigmoid(self._discrim_logits)
-    return train_reward
+  def build_graph(self):
+    inputs = util.build_inputs(self._observation_space, self._action_space,
+                               scale=self._scale)
+    self._obs_ph, self._act_ph, self._next_obs_ph = inputs[:3]
+    self.obs_inp, self.act_inp, self.next_obs_inp = inputs[3:]
 
-  def build_disc_loss(self):
-    super().build_disc_loss()
+    with tf.variable_scope("discrim_network"):
+      self._discrim_mlp, self._discrim_logits = self._build_discrim_net(
+          self.obs_inp, self.act_inp)
+    self._policy_test_reward = self._policy_train_reward \
+        = -tf.log_sigmoid(self._discrim_logits)
 
-    disc_loss = tf.nn.sigmoid_cross_entropy_with_logits(
+    self._disc_loss = tf.nn.sigmoid_cross_entropy_with_logits(
         logits=self._discrim_logits,
-        labels=tf.cast(self.labels_ph, tf.float32),
-    )
-
-    return disc_loss
+        labels=tf.cast(self.labels_ph, tf.float32))
+
+    # TODO(sam): push all this stuff into DiscrimNetwork, since it's not
+    # GAIL-specific (only diff is in AIRL we need to convert softmax logits to
+    # Bernoulli logits first; just do bern_logits = softmax_logits[:, 1] -
+    # softmax_logits[:, 0]).
+
+    # Compute debugging stats. For labels, expert (exp) = 0, and generated
+    # (gen) = 1.
+    bin_is_generated_pred = self._discrim_logits > 0
+    bin_is_generated_true = self._labels_ph > 0
+    bin_is_expert_true = tf.logical_not(bin_is_generated_true)
+    int_is_generated_pred = tf.cast(bin_is_generated_pred, tf.int32)
+    int_is_generated_true = tf.cast(bin_is_generated_true, tf.int32)
+    n_generated = tf.reduce_sum(int_is_generated_true)
+    n_labels = tf.size(self._labels_ph)
+    n_expert = n_labels - n_generated
+    pct_expert = tf.cast(n_expert, tf.float32) / tf.cast(n_labels, tf.float32)
+    n_expert_pred = tf.size(bin_is_generated_pred) \
+        - tf.reduce_sum(int_is_generated_pred)
+    pct_expert_pred = tf.cast(n_expert_pred, tf.float32) \
+        / tf.cast(n_labels, tf.float32)
+    correct_vec = tf.equal(bin_is_generated_pred, bin_is_generated_true)
+    acc = tf.reduce_mean(tf.cast(correct_vec, tf.float32))
+    expert_acc = tf.reduce_sum(tf.cast(
+      tf.logical_and(bin_is_expert_true, correct_vec), tf.float32)) \
+        / tf.cast(tf.maximum(1, n_expert), tf.float32)
+    generated_acc = tf.reduce_sum(tf.cast(
+      tf.logical_and(bin_is_generated_true, correct_vec), tf.float32)) \
+        / tf.cast(tf.maximum(1, n_generated), tf.float32)
+    label_dist = tf.distributions.Bernoulli(logits=self._discrim_logits)
+    entropy = tf.reduce_mean(label_dist.entropy())
```

Comment:
Thanks for adding all of these summary stats, they look like useful items for hp tuning / debugging